dvc/repo/experiments/show.py

```diff
@@ -100,11 +100,10 @@ def _collect_experiment_commit(
                 for refspec in ["refs/tags", "refs/heads"]:
                     name = repo.scm.describe(rev, base=refspec)
                     if name:
+                        name = name.replace(f"{refspec}/", "")
                         break
-            if not name:
-                name = repo.experiments.get_exact_name(rev)
+            name = name or repo.experiments.get_exact_name(rev)
             if name:
-                name = name.rsplit("/")[-1]
```

Comment:
I believe this was not really needed after the introduction of `repo.experiments.get_exact_name(rev)` in https://github.com/iterative/dvc/pull/6382 but it was left here, causing the unnecessary truncation.

Reply:
There was no test case for it